### PR TITLE
Alex and serena/s1 f1 groundwork schemas

### PR DESCRIPTION
**Pull request by alexb8z**
Groundwork schemas complete for UserState and ChatbotMessage. Selected types for fields, but these can be changed later if needed (for example, if an object would work better than an array for images or something like that).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,9 +30,12 @@
     "typescript": "^3.3.3"
   },
   "dependencies": {
+    "@types/mongoose": "^5.10.3",
     "@types/qs": "^6.9.5",
     "@types/twilio": "^2.11.0",
     "azure-functions-core-tools": "^3.0.3233",
+    "mongoose": "^5.11.14",
+    "n": "^7.0.1",
```
Comment (Arthelon) [inline package.json]: Don't think this dependency was meant to be included (probably meant to install it globally)

File: package.json [hunk 2]
```diff
@@ -30,9 +30,12 @@
     "typescript": "^3.3.3"
   },
   "dependencies": {
+    "@types/mongoose": "^5.10.3",
```
Comment (Arthelon) [inline package.json]: all `@types` packages should be installed as devDependencies

File: models/ChatbotMessage.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const ChatbotMessageSchema = new Schema({
+  body: { type: String, required: true },
+  images: { type: Array, required: false },
+  nextMessages: { type: Array, required: true },
```
Comment (Fattimo) [inline models/ChatbotMessage.ts]: https://mongoosejs.com/docs/schematypes.html#maps

Great work on the rest of this PR, everything looks really good. One small adjustment here: I talked to Alex about this issue during our 1-1, but I was thinking about we should go about actually splitting into the correct message. An array is one way to do it, but we'd somehow need to transmit and store a correct index, which is possible in some way, but seems very clunky as I think about it more.

I think the best way to do it is through a map (or object, but it's referred to as a map in Mongoose documentation oops) which I've linked above. Additionally, be sure to type the content of arrays as done on line 54 in this previous project here: https://github.com/GTBitsOfGood/gen-soln/blob/develop/server/models/nonprofit.js using `type: Schema.ObjectId, ref: 'ChatbotMessage'`.

File: models/ChatbotMessage.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  previousMessage: { type: String, required: true },
```
Comment (Fattimo) [inline models/ChatbotMessage.ts]: Type this as a reference to the key of another ChatbotMessage, same as explained above

File: models/ChatbotMessage.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const ChatbotMessageSchema = new Schema({
+  body: { type: String, required: true },
+  images: { type: Array, required: false },
```
Comment (Fattimo) [inline models/ChatbotMessage.ts]: Type this array as explained below

File: models/UserState.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const UserStateSchema = new Schema({
+  userId: { type: String, required: true, unique: true },
+  currMessage: { type: String, required: true },
```
Comment (Fattimo) [inline models/UserState.ts]: Type this as a key to a message as explained above

**Review by Fattimo — CHANGES_REQUESTED** [review]

**Review by Fattimo — APPROVED** [review]
Great job!

**Pull request merged**